src/EventStore.Client/SubscriptionPosition.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+
+#nullable enable
+namespace EventStore.Client {
+	/// <summary>
+	/// A structure representing the logical position of a subscription to all. />
+	/// </summary>
+	public readonly struct SubscriptionPosition : IEquatable<SubscriptionPosition>, IComparable<SubscriptionPosition> {
```

Comment:
we could consider implementing non-generic `IComparable` too https://github.com/EventStore/EventStore-Client-Dotnet/pull/111